Rofl.Reader/Parsers/RoflParser.cs

```diff
@@ -16,14 +16,33 @@ public class RoflParser : IReplayParser
     {
         private readonly string exceptionOriginName = "RoflParser";
         private readonly byte[] _magicNumbers = new byte[] { 0x52, 0x49, 0x4F, 0x54 };
+        private const int lengthFieldOffset = 262;
+        private const int lengthFieldByteSize = 26;
+        private FileStream fileStream;
+        private LengthFields lengthFields;
 
-        public async Task<ReplayHeader> ReadReplayAsync(FileStream fileStream)
+        public async Task<ReplayHeader> ReadReplayAsync(FileStream fs)
+        {
+            this.fileStream = fs;
+            await RunChecksAsync();
```

Comment:
The `RunChecksAsync` function serves to group the pre-checks. As it only groups together two functions for now - yes, this function call may feel a bit redundant.
However, once somebody decides to add another pre-check, it would make sense to call this grouping function instead of calling each check seperately.

It also supports code readability by dividing the process into clearly separated steps:
1. Check if conditions are met
2. Read from file
3. Parse info